lessons/micropython/mini-workshop/index.md

```diff
@@ -0,0 +1,319 @@
+
+>>> 3 * 3
+
+>>> 1 / 4
+
+>>> (84 + 5) * 253
+```
+
+
+## Světýlko
+
+MicroPython na nach destičkách má připravený objekt `led`, jehož metodou
+`value` jde ovládat svítící dioda.
+Stačí jen zadat správné příkazy:
+
+```pycon
+>>> led.value(1)
+>>> led.value(0)
+```
+
+
+## Tlačítko
+
+K destičce je připojeno i tlačítko.
+Metodou `btn.value()` se zeptáš, jestli je právě zmáčknuté nebo ne.
+
+```pycon
+>>> btn.value()
+```
+
+Zkus tenhle příkaz zadat, zatímco tlačítko držíš.
+
+
+## Výrazy
+
+Informace o tom, jestli je tlačítko zmáčknuté – `btn.value()` – je číslo:
+buď 0 nebo 1.
+Výraz `btn.value()` můžeš použít kdekoliv, kde se objevuje číslo
+0 nebo 1 – třeba pro rozsvícení nebo zhasnutí diody:
+
+```pycon
+>>> led.value(btn.value())
+```
+
+
+## Cyklus
+
+Existuje způsob, jak nechat MicroPython něco opakovat stále dokola:
+takzvaný *nekonečný cyklus*.
+Na jeden řádek napiš `while True:` (s velkým T a dvojtečkou na konci),
+a potom *tělo cyklu* – příkaz, který se má provést.
+MicroPython tělo cyklu automaticky *odsadí* – přidá na začátek řádku
+mezery. Ty mezery nemaž.
+
+Nakonec několikrát stiskni Enter, dokud se program nespustí.
+
+```pycon
+>>> while True:
+...     led.value(btn.value())
+...
+...
+```
+
+Nekonečný cyklus se opakuje donekonečna.
+Je ale způsob jak ho přerušit, až tě omrzí: zmáčkni Ctrl+C.
+
+<div style="page-break-after: always;"></div>
+
+## Motorek
+
+Další objekt, který můžeš použít, je `servo`.
+Ten ovládá servomotor, který můžeš nastavit na konkrétní úhel. Třeba:
+
+```pycon
+>>> servo.duty(120)
+>>> servo.duty(40)
+```
+
+... nebo něco mezi tím.
+
+Nepoužívej prosím čísla menší než 40 nebo větší než 120, motorek se tím může
+ničit.
+
+
+## Matematika
+
+Před chvílí jsme si ukázali tento kód:
+
+```pycon
+>>> led.value(btn.value())
+```
+
+Hodnota `btn.value()` může být buď 1 nebo 0.
+Pojďme ji použít pro motorek: chceme, aby se natočil buď na `40` nebo na
+`120`, podle toho, jestli je tlačítko zmáčknuté.
+Použij trochu matematiky:
+
+    40 + 0 * 80 = 40
+    40 + 1 * 80 = 120
+
+a dostaneš následující kód:
+
+```pycon
+>>> servo.duty(40 + btn.value() * 80)
+```
+
+Změníš program tak, aby se natáčel na `50` nebo `90`?
+
+Dokážeš výsledek dát do cyklu, aby program na zmáčknutí či puštění tlačítka
+reagoval automaticky?
+
+
+## Barvy
+
+Poslední věc připojená k destičce je LED pásek s *několika* světýlkama.
+Každé z nich se dá nastavit na nějakou barvu.
+Zkus si to:
+
+```pycon
+>>> strip[0] = RED
+>>> strip[1] = GREEN
+>>> strip[2] = BLUE
+>>> strip.write()
+```
+
+První tři příkazy připravují jednotlivé barvy, a poslední všechno připravené
+pošle do LED pásku.
+Až budeš experimentovat, nezapomeň na `strip.write()` – bez toho
+se barvy neukážou.
+
+<div style="page-break-after: always;"></div>
+
+## Program
+
+<img src="{{ static('icon_gedit.png')}}" style="float:right;height:5em;">
+
+Teď si přestaneme jen tak hrát, a začneme psát program.
+
+Spusť *editor*: Klikni na *Activities*; potom vyber z levého panelu Gedit
```

Comment:
Mám tam v anglické variantě napsané Text editor, ne Gedit. Změnil bych to na *Textový editor* (za předpokladu, že to bude všechno česky)